internal/common/compose_result.go

```diff
@@ -6,32 +6,33 @@ import (
 	"io"
 )
 
+type Assembler struct {
```

Comment:
I just realized we possibly don't want to have these helper types public. Can you make Assembler`, `Stage` and `Build` private? They are not really meant to be used on their own. Sorry for not pointing earlier.

Reply:
No worries, changed that. Thanks!